app/components/App.js

```diff
@@ -25,13 +25,12 @@ const App = ({ api, widgets, isSyncing }) => {
     setPanelVisible(true)
   }, [setPanelVisible])
 
-  const handlePanelSubmit = useCallback(cId => {
+  const handlePanelSubmit = useCallback(async widgetObject => {
     setPanelVisible(false)
-
-    // TODO: adapt contract to remove priority
-    const priority = toHex(widgets.length)
-    api.addWidget(priority, cId).toPromise()
-  }, [ setPanelVisible, widgets.length ])
+    widgets.push(widgetObject)
+    const cId = (await ipfs.dag.put(widgets, { pin: true })).toBaseEncodedString()
```

Comment:
Better define local state:

```suggestion
    const nextWidgets = [...widgets, widgetObject]
    const cId = (await ipfs.dag.put(nextWidgets, { pin: true })).toBaseEncodedString()
```